src/main/java/com/google/neighborgood/servlets/MessageServlet.java

```diff
@@ -0,0 +1,109 @@
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.neighborgood.Message;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that loads and records new message entities. */
+@WebServlet("/messages")
+public class MessageServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String taskKey = request.getParameter("key");
+    if (taskKey == null) {
+      System.err.println("No task key string provided");
+    }
+
+    // Make sure that the user has already logged into his account
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect(userService.createLoginURL("/account.jsp"));
+      return;
+    }
+
+    Filter filter = new FilterPredicate("taskId", FilterOperator.EQUAL, taskKey);
+    Query query =
+        new Query("Message").setFilter(filter).addSort("sentTime", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Message> messages = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      messages.add(new Message(entity));
+    }
+
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // First check whether the user is logged in
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect(userService.createLoginURL("/account.jsp"));
+      return;
+    }
+
+    // Get the keyString of the task
+    String keyString = request.getParameter("task-id");
+    if (keyString == null) {
+      System.err.println("The task id is not included");
+      return;
+    }
+
+    // Get the message content
+    String message = request.getParameter("msg");
+    if (message == null || message.trim().equals("")) {
+      System.err.println("The input message is empty");
+      return;
+    }
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Create a message entity to store the related information
+    Entity msgEntity = new Entity("Message");
+    msgEntity.setProperty("message", message);
+    msgEntity.setProperty("taskId", keyString);
```

Comment:
Same comment as before. I would suggest wording this as either id or key, instead of using both word choices.

Reply:
I would use id for these variables in the next commit.